### ISSUE TITLE
Themes from ~/.config/mintty(/themes) no longer listed in options dialog

### ISSUE THREAD

**Issue by rb83** [mintty/mintty#1016]
Using mintty 3.1.8. Only themes from `/usr/share/mintty(/themes)` are listed.  `mintheme -l` correctly lists them all (and allows switching to them). Of note, `~/.config/mintty` on my setup is a softlink. This has worked until recently (I can't recall the previously used version from the top of my head, but it must be the version packaged by cygwin preceding 3.1.8).

Tried to hunt down where it goes wrong in the code, but C ain't really my strong suit, and reading `get_resource_file` it can't really see what's going wrong.

**Comment by mintty** [comment]
This may be an interworking problem between cygwin links and Windows links, there have been recent changes in cygwin. No recent changes in mintty are related.
Can you try to downgrade your cygwin version?

**Comment by mintty** [comment]
What kind of softlink is your mintty config dir? What does `ls -l ~/.config` say?

**Comment by rb83** [comment]
> What kind of softlink is your mintty config dir? What does `ls -l ~/.config` say?

It's a relative link within my home dir:
```
$ ls -l ~/.config
(...snip...)
lrwxrwxrwx  1 (redacted) (redacted)   32 15. Jun 10:36 mintty -> ../repos/dotfiles/.config/mintty
```

> Can you try to downgrade your cygwin version?

Will attempt this, or will try to reproduce an older version within a VM, as the issue occurs on my daily driver.

**Comment by rb83** [comment]
> This may be an interworking problem between cygwin links and Windows links, there have been recent changes in cygwin.

Come to think of it, links were created with GNU's `stow` and not with `ln`. Might be related.

**Comment by mintty** [comment]
> Come to think of it, links were created with GNU's stow and not with ln. Might be related.

Absolutely. Please try a quick cross-check, replacing the link with a cygwin link.

**Comment by rb83** [comment]
Solved it:

Recreating the link with `ln` does not help; upon reading https://cygwin.com/cygwin-ug-net/using.html#pathnames-symlinks in recent cygwin versions, this attempts to create windows-native symlinks (using NTFS-specific reparse points) first, before falling back to creating plain files containing a magic cookie. Trying to force native windows symlinks errors out in my environment (this may be due to not being on the most recent windows release):

```
$ export CYGWIN=winsymlinks:nativestrict
$ ln -s ../repos/dotfiles/.config/mintty
ln: failed to create symbolic link './mintty': Operation not permitted
```

By forcing cygwin to use windows' shortcuts instead of symlinks, it works again:
```
$ export CYGWIN=winsymlinks # or winsymlinks:lnk
$ ln -s ../repos/dotfiles/.config/mintty
(themes appear again)
```

I'm not sure wether old cygwin versions have preferred shortcuts to cygwin magic plain files; and wether this behaviour has been flipped around with the introduction (and priorization of) windows-native symlinks. 

I'm not sure if it's within mintty's scope to handle cygwin magic plain files for config directories?

**Comment by mintty** [comment]
> I'm not sure if it's within mintty's scope to handle cygwin magic plain files for config directories?

This should certainly work implicitly. Mintty opens resource files with POSIX functions, and those should work on links created with cygwin ln. It's quite weird that you could fix it by switching to CYGWIN=winsymlinks.

**Comment by mintty** [comment]
I have to modify my previous statement:

> Mintty opens resource files with POSIX functions

Yes, but it lists available resources using Windows functions. That's most likely the problem.
While there was a reason to do it like this, I'll check how to change it.

**Comment by k-takata** [comment]
To create windows native symlinks, you need to enable Windows 10's developer mode, or grant admin permissions (by UAC).
One pitfall about windows native symlinks is that a symlink itself has a directory attribute (unlike linux), and if you wrongly set (or unset) the attribute, the symlink won't work.

**Comment by mintty** [comment]
Sorry, I related a series of recent commits to this issue in mistake.

**Comment by mintty** [comment]
Released 3.3.0.
